File: selection sorting/selection_sorting.dart
class SelectionSorting {
  void selectionSorting(List<int> arr) {
    for (int i = 0; i < arr.length; i++) {
      int minIndex = i;
      for (int j = i; j < arr.length; j++) {
        if (arr[j] < arr[i]) {
          minIndex = j;
        }
      }
      final minValue = arr.removeAt(minIndex);
      arr.insert(i, minValue);
    }
  }
}

void main() {
  List<int> arr = [4, 8, 1];
  final selectFun = SelectionSorting();

  selectFun.selectionSorting(arr);

  print(arr);

// select.selectionSorting(arr);
}

File: selection sorting/selection_sorting.py

def selection_sort(list):
    n = len(list)

    for i in range(n):
        min_index = i
        for j in range(i,n):
            if(list[j]< list[i]):
                min_index = j
        min_value =list.pop(min_index)
        list.insert(i,min_value)
    
    
    
arr = [4,3,1]
selection_sort(arr)  
print(arr)          